build-logic/conventions/src/main/kotlin/com/squareup/anvil/conventions/LibraryPlugin.kt

```diff
@@ -0,0 +1,49 @@
+package com.squareup.anvil.conventions
+
+import com.rickbusarow.kgx.libsCatalog
+import com.rickbusarow.kgx.pluginId
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.plugins.ide.idea.model.IdeaModel
+
+open class LibraryPlugin : Plugin<Project> {
+  override fun apply(target: Project) {
+
+    target.plugins.apply(target.libsCatalog.pluginId("kotlin-jvm"))
+    target.plugins.apply(KtlintConventionPlugin::class.java)
+
+    when (target.path) {
+      ":annotations",
+      ":annotations-optional",
+      ":compiler",
+      ":compiler-api",
+      ":compiler-utils",
+      ":gradle-plugin",
+      -> target.setBuildDir()
```

Comment:
Conditionally applying by target path/module like this feels unexpected instead of setting the build dir for anyone that has chosen to apply this plugin. What do you think about having a separate plugin to handle the case for simple lib modules that only want the above (kotlin-jvm, ktlint) plugins applied? The simpler plugin could be something like `MinimalLibraryPlugin` or the other direction where the library plugin with this build dir setup logic could be named around what it's doing differently to indicate which modules should use it.